agent/util/archive_tar.go

```diff
@@ -29,6 +30,10 @@ FileLoop:
 		if ctx.Err() != nil {
 			return numFilesArchived, errors.Wrap(ctx.Err(), "timeout when building archive")
 		}
+		if reflect.ValueOf(file.Info).IsNil() {
```

Comment:
If we know what type of FileInfo we expect back we could instead do a regular type assertion, but otherwise I think this is fine

Reply:
That's true, but I worry that it's not guaranteed to not change by Go.